control/server.py

```diff
@@ -113,6 +113,11 @@ def serve(self):
 
         # Start server
         self.server.start()
+        try:
+            rpc_nvmf.nvmf_delete_subsystem(self.spdk_rpc_ping_client, "nqn.2014-08.org.nvmexpress.discovery")
```

Comment:
SPDK has already defined the discovery NQN specified in the NVMe-over-Fabrics specification, see [here](https://github.com/spdk/spdk/blob/396e6facb307e108eb454c9e0769a2b108ff2f0d/python/spdk/rpc/nvmf.py#L301-L302)
```suggestion
            rpc_nvmf.nvmf_delete_subsystem(self.spdk_rpc_ping_client, "discovery")
```

Reply:
rpc_nvmf.nvmf_delete_subsystem(self.spdk_rpc_ping_client, "discovery")
ceph-nvmeof-nvmeof-1  |   File "/usr/lib/python3.9/site-packages/spdk/rpc/nvmf.py", line 500, in nvmf_delete_subsystem
ceph-nvmeof-nvmeof-1  |     return client.call('nvmf_delete_subsystem', params)
ceph-nvmeof-nvmeof-1  |   File "/usr/lib/python3.9/site-packages/spdk/rpc/client.py", line 203, in call
ceph-nvmeof-nvmeof-1  |     raise JSONRPCException(msg)
ceph-nvmeof-nvmeof-1  | spdk.rpc.client.JSONRPCException: request:
ceph-nvmeof-nvmeof-1  | {
ceph-nvmeof-nvmeof-1  |   "nqn": "discovery",
ceph-nvmeof-nvmeof-1  |   "method": "nvmf_delete_subsystem",
ceph-nvmeof-nvmeof-1  |   "req_id": 1
ceph-nvmeof-nvmeof-1  | }
ceph-nvmeof-nvmeof-1  | Got JSON-RPC error response
ceph-nvmeof-nvmeof-1  | response:
ceph-nvmeof-nvmeof-1  | {
ceph-nvmeof-nvmeof-1  |   "code": -32602,
ceph-nvmeof-nvmeof-1  |   "message": "Invalid parameters"
ceph-nvmeof-nvmeof-1  | }

Reply:
Oh, it looks like [nvmf_delete_subsystem](https://github.com/spdk/spdk/blob/396e6facb307e108eb454c9e0769a2b108ff2f0d/python/spdk/rpc/nvmf.py#L488-L503) doesn't handle translation. This seems to be a feature exclusive to `nvmf_subsystem_remove_listener`. Thank you for verifying this.